domains/transfer-agreements/TransferAgreements.API/API/Controllers/TransferAgreementsController.cs

```diff
@@ -20,16 +27,18 @@ public class TransferAgreementsController : ControllerBase
     private readonly ITransferAgreementRepository transferAgreementRepository;
     private readonly IValidator<CreateTransferAgreement> createTransferAgreementValidator;
     private readonly IWalletDepositEndpointService walletDepositEndpointService;
-
+    private readonly IOptions<ProjectOriginOptions> projectOriginOptions;
 
     public TransferAgreementsController(
         ITransferAgreementRepository transferAgreementRepository,
         IValidator<CreateTransferAgreement> createTransferAgreementValidator,
-        IWalletDepositEndpointService walletDepositEndpointService)
+        IWalletDepositEndpointService walletDepositEndpointService,
```

Comment:
I think there is some odd with this now that a service layer is introduced.
We have this controller that has a dependency to the `Data` layer and a `Service` layer now.
Have you considered making a transferagreement service that will handle everything regarding transfer agreements data layer and the wallet deposit endpoint?

Reply:
No

Reply:
I will create a zenhub issue with a tech-debt label, for refacturing the transfer agreement domain.